### ISSUE TITLE
EmptyThrowable "Invalid range specified"

### ISSUE THREAD

**Issue by btilbrook-nextfaze** [cheptsov/AdvancedExpressionFolding#57]
IDE: Android Studio 2.4 Preview 7
OS: Mac OS X 10.11.6 (15G1421)

Stacktrace:
```
Invalid range specified: (17617,0); 
com.intellij.openapi.diagnostic.Logger$EmptyThrowable
	at com.intellij.openapi.diagnostic.Logger.error(Logger.java:134)
	at com.intellij.openapi.util.TextRange.assertProperRange(TextRange.java:222)
	at com.intellij.openapi.util.TextRange.assertProperRange(TextRange.java:217)
	at com.intellij.openapi.util.TextRange.assertProperRange(TextRange.java:213)
	at com.intellij.openapi.util.TextRange.<init>(TextRange.java:43)
	at com.intellij.openapi.util.TextRange.<init>(TextRange.java:32)
	at com.intellij.openapi.editor.DocumentFragment.<init>(DocumentFragment.java:26)
	at com.intellij.advancedExpressionFolding.AdvancedExpressionFoldingHighlightingComponent.createDocumentFragment(AdvancedExpressionFoldingHighlightingComponent.java:202)
	at com.intellij.advancedExpressionFolding.AdvancedExpressionFoldingHighlightingComponent.access$200(AdvancedExpressionFoldingHighlightingComponent.java:33)
	at com.intellij.advancedExpressionFolding.AdvancedExpressionFoldingHighlightingComponent$3.mouseMoved(AdvancedExpressionFoldingHighlightingComponent.java:138)
	at com.intellij.openapi.editor.impl.EditorImpl$MyMouseMotionListener.mouseMoved(EditorImpl.java:3925)
	at java.awt.Component.processMouseMotionEvent(Component.java:6586)
	at javax.swing.JComponent.processMouseMotionEvent(JComponent.java:3342)
	at java.awt.Component.processEvent(Component.java:6310)
	at java.awt.Container.processEvent(Container.java:2236)
	at java.awt.Component.dispatchEventImpl(Component.java:4897)
	at java.awt.Container.dispatchEventImpl(Container.java:2294)
	at java.awt.Component.dispatchEvent(Component.java:4719)
	at java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4888)
	at java.awt.LightweightDispatcher.processMouseEvent(Container.java:4538)
	at java.awt.LightweightDispatcher.dispatchEvent(Container.java:4466)
	at java.awt.Container.dispatchEventImpl(Container.java:2280)
	at java.awt.Window.dispatchEventImpl(Window.java:2746)
	at java.awt.Component.dispatchEvent(Component.java:4719)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:764)
	at java.awt.EventQueue.access$500(EventQueue.java:98)
	at java.awt.EventQueue$3.run(EventQueue.java:715)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:80)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:90)
	at java.awt.EventQueue$4.run(EventQueue.java:737)
	at java.awt.EventQueue$4.run(EventQueue.java:735)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:80)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:734)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:827)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:651)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:365)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
```

**Comment by cheptsov** [comment]
I've just published 0.8.2 (EAP) with some changes addressing this problem too. Could you please check if the problem still exists?

**Comment by btilbrook-nextfaze** [comment]
I wasn't able to reproduce it initially, and don't like my chances now. Might as well close.

Cheers for the fixes!
